src/util.js

```diff
@@ -276,6 +276,45 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+var _hookIdCounter = 0;
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
```

Comment:
OK, this problem was solved with current implementation, I have checked its performance with same benchmark, it is same like manually wrapping functions (but manual wrapping does not have support for unhook).